File: .ProgramingBasics/Simple-Loops/11.OddEvenPosition/src/OddEvenPosition.java
import java.text.MessageFormat;
import java.util.Scanner;

public class OddEvenPosition {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double n = Double.parseDouble(scanner.nextLine());
        double oddSum = 0;
        double oddMin = 1000000000.0;
        double oddMax = -1000000000.0;
        double evenSum = 0;
        double evenMin = 1000000000.0;
        double evenMax = -1000000000.0;

        for (int i = 1; i <= n; i++) {
            double num = Double.parseDouble(scanner.nextLine());

            if (i % 2 == 0) {
                evenSum += num;
                evenMin = Math.min(evenMin, num);
                evenMax = Math.max(evenMax, num);
            }
            else {
                oddSum += num;
                oddMin = Math.min(oddMin, num);
                oddMax = Math.max(oddMax, num);
            }
        }

        System.out.println(MessageFormat.format("OddSum={0}",oddSum));

        if (oddMin == 1000000000.0) {
            System.out.println("OddMin=No");
        }
        else {
            System.out.println(MessageFormat.format("OddMin={0}",oddMin));
        }

        if (oddMax == -1000000000.0) {
            System.out.println("OddMax=No");
        }
        else {
            System.out.println(MessageFormat.format("OddMax={0}", oddMax));
        }

        System.out.println(MessageFormat.format("EvenSum={0}", evenSum));

        if (evenMin == 1000000000.0) {
            System.out.println("evenMin=No");
        }
        else {
            System.out.println(MessageFormat.format("evenMin={0}", evenMin));
        }

        if (evenMax == -1000000000.0) {
            System.out.println("evenMax=No");
        }
        else {
            System.out.println(MessageFormat.format("evenMax={0}", evenMax));
        }
    }
}

File: ProgramingBasics/SimpleLoops/11.OddEvenPosition/11.OddEvenPosition/OddEvenPosition.cs
using System;

namespace _11.OddEvenPosition
{
    public class OddEvenPosition
    {
        public static void Main(string[] args)
        {
            var n = double.Parse(Console.ReadLine());
            var oddSum = 0d;
            var oddMin = 1000000000.0;
            var oddMax = -1000000000.0;
            var evenSum = 0d;
            var evenMin = 1000000000.0;
            var evenMax = -1000000000.0;

            for (var i = 1; i <= n; i++)
            {
                var num = double.Parse(Console.ReadLine());

                if (i % 2 == 0)
                {
                    evenSum += num;
                    evenMin = Math.Min(evenMin, num);
                    evenMax = Math.Max(evenMax, num);
                }
                else
                {
                    oddSum += num;
                    oddMin = Math.Min(oddMin, num);
                    oddMax = Math.Max(oddMax, num);
                }
            }

            Console.WriteLine($"OddSum={oddSum}");

            if (oddMin == 1000000000.0)
            {
                Console.WriteLine("OddMin=No");
            }
            else
            {
                Console.WriteLine($"OddMin={oddMin}");
            }

            if (oddMax == -1000000000.0)
            {
                Console.WriteLine("OddMax=No");
            }
            else
            {
                Console.WriteLine($"OddMax={oddMax}");
            }

            Console.WriteLine($"EvenSum={evenSum}");

            if (evenMin == 1000000000.0)
            {
                Console.WriteLine("evenMin=No");
            }
            else
            {
                Console.WriteLine($"evenMin={evenMin}");
            }

            if (evenMax == -1000000000.0)
            {
                Console.WriteLine("evenMax=No");
            }
            else
            {
                Console.WriteLine($"evenMax={evenMax}");
            }
        }
    }
}